sdk/core/azure-core/src/main/java/com/azure/core/http/rest/BinaryDataResponse.java

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.core.http.rest;
+
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.util.BinaryData;
+
+import java.io.Closeable;
+
+/**
+ * REST response with a streaming content.
+ */
+public final class BinaryDataResponse extends SimpleResponse<BinaryData> implements Closeable {
```

Comment:
Do we actually need this?

Reply:
I think so.
I.e. this is supposed to be generalized version of `com.azure.core.http.rest.StreamResponse`.
Plus `SimpleResponse` isn't `Closeable` and we have to return a type that's closeable if we're returning open network stream in some form.

Reply:
we can probably do some tricks with `StreamResponse` though to get this functionality there and not create an extra type.
I can look at such possibility later.